gradle.properties

```diff
@@ -1 +1 @@
-version=4.0.1
```

Comment:
Doesn't look like the right version number, also there's a release script which will bump the version number everywhere in the repo

Reply:
Role_Name 

[SpringBootTelemetryInitializer](https://github.com/Microsoft/ApplicationInsights-Java/blob/bc01d832bb351c5150978b816cf394de24853aea/azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/initializer/SpringBootTelemetryInitializer.java) takes care of setting rolename as they seem to have deprecated DeviceContext. setRoleName (  https://github.com/Microsoft/ApplicationInsights-Java/blob/bc01d832bb351c5150978b816cf394de24853aea/core/src/main/java/com/microsoft/applicationinsights/extensibility/context/DeviceContext.java)  with new CloudContext.

Reply:
Sure, I will raise a PR for SyntheticHeaders.

And AbstractAppInsights.java is being used across many projects using java-logging. I checked with @doncem  last week and learnt that it was there for the purpose of having ready made features available like version. Happy to get rid of it/ deprecate if everyones agrees.

Reply:
keep it if it gives value, but not convinced it is currently...

Reply:
should deprecate for sure. starter should deal with all the checks and customisations. introduce custom initialiser if needed

Reply:
Raised https://github.com/Microsoft/ApplicationInsights-Java/pull/813 to be able to deprecate SyntheticHeaders.java

Reply:
i meant deprecating `AbstractAppInsights`. Definitely not needed and can be just simply converted to custom initialiser

Reply:
I am merging that in a while,  wanted to be able to deprecate SyntheticHeaders as per Tim's comment above.

Reply:
sounds good 👍 hopefully they take the PR

Reply:
Used release script to bump up the version . Closing this PR for now and will open another one if Microsoft accepts the PR and releases a  version to deprecate SyntheticHeaders